apps/client/src/common/stores/clientStore.ts

```diff
@@ -0,0 +1,49 @@
+import { create } from 'zustand';
+
+interface ClientStore {
+  self?: string;
+  setName: (newValue: string) => void;
+
+  identify: Record<string, boolean>;
+  setIdent: (clientName: string, value: boolean) => void;
+}
+
+const clientNameKey = 'ontime-client-name';
+
+function persistNameInStorage(newValue: string) {
+  localStorage.setItem(clientNameKey, newValue);
+}
+
+export const useClientStore = create<ClientStore>((set) => ({
+  self: localStorage.getItem(clientNameKey) ?? undefined,
```

Comment:
why casting to undefined? could we not deal with the null?

Reply:
don't think I wrote this part, but yeah null wold be OK I think